src/cliframe/cliReducer.ts

```diff
@@ -0,0 +1,61 @@
+import * as Messaging from "@src/lib/messaging"
+import * as Complete from "@src/lib/completefns"
+import { TriCliAction, FocusIndexAction, FocusDirectionAction } from "@src/cliframe/actionTypes"
+import { TriCliState } from "@src/cliframe/store"
+
+export function rootCliReducer(state: TriCliState, action: TriCliAction): TriCliState {
```

Comment:
How did we get to a place where the State Of The Art in JavaScript says

```
nextState = action(prevState) // bad

function reducer(state,not_quite_the_action) {
... massive switch statement of all possible actions ...
case "action": return action(state)
}
nextState = reducer(prevState,"action") // much better

```

Am I missing something here? Is it necessary because JS doesn't have types by default, so it doesn't know which functions can be applied to states?

/rant